src/pages/incidents/[id]/index.tsx

```diff
@@ -46,22 +51,53 @@ const INCIDENT_QUERY = `
 
 const Incident: Component = () => {
   const params = useParams()
+  const [incident, setIncident] = createSignal<IncidentType>()
   const [incidentResult, incidentState] = createQuery({
     query: INCIDENT_QUERY,
     variables: { id: params.id },
   })
-  const incident = () => ({
-    ...incidentResult()?.incident,
-    ...incidentResult()?.incidentEventTime,
+
+  createEffect(() => {
+    setIncident({
+      ...incidentResult()?.incident,
+      ...incidentResult()?.incidentEventTime,
+    })
   })
+  //handleCreateEvent
+  const handleOnUpdate = ({ event, statusMessage }: IncidentUpdateProps) => {
+    if (incident()) {
+      const prevIncident = incident() as IncidentType
+
+      if (event) {
+        prevIncident.events = [...incidentResult()?.incident.events, event]
+      }
+
+      if (statusMessage) {
+        prevIncident.statusMessage = [
+          ...incidentResult()?.incident.statusMessage,
+          statusMessage,
+        ]
+      }
+
+      setIncident(prevIncident)
+    }
+  }
+
+  // signal: incident
+  // create effect sets the incident after fetch
+  // handler functions for messages, events that update incident signal
+  // add date to event
 
   return (
     <AppLayout>
       <Show when={!incidentState().fetching} fallback={<p>Loading...</p>}>
```

Comment:
I think if we change this, then we don't have to handle the `undefined` state for `incident` on the child components. That's preferable.

```suggestion
      <Show when={incident()} fallback={<p>Loading...</p>}>
```